scripts/feature_tests/upgrade/controlplane_upgrade/3cp_1w_k8sVer_bootDiskImage_scaleInWorker_upgrade.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+set -x
+
+# shellcheck disable=SC1091
+source ../common.sh
+
+echo '' >~/.ssh/known_hosts
+
+start_logging "${1}"
+
+set_number_of_master_node_replicas 3
+set_number_of_worker_node_replicas 1
+
+provision_controlplane_node
+
+CLUSTER_NAME=$(kubectl get clusters -n metal3 | grep Provisioned | cut -f1 -d' ')
+
+wait_for_ctrlplane_provisioning_start
+
+CP_ORIGINAL_NODE_LIST=$(kubectl get bmh -n metal3 | grep control | grep -v ready | cut -f1 -d' ')
+echo "BareMetalHosts ${CP_ORIGINAL_NODE_LIST} are in provisioning or provisioned state"
+
+NODE_IP_LIST=()
+for node in "${CP_ORIGINAL_NODE_LIST[@]}"; do
+  #shellcheck disable=SC2179
+  NODE_IP_LIST+=$(sudo virsh net-dhcp-leases baremetal | grep "${node}" | awk '{{print $5}}' | cut -f1 -d'/')
+done
+# shellcheck disable=SC2145
+echo "NODE_IP_LIST ${NODE_IP_LIST[@]}"
+wait_for_ctrlplane_provisioning_complete "${CP_ORIGINAL_NODE_LIST[@]}" "${NODE_IP_LIST[@]}"
+
+# list consists of 3 ctrl plane node ips
+CP_IP="${NODE_IP_LIST[0]}"
+
+# apply CNI
+ssh -o PasswordAuthentication=no -o "StrictHostKeyChecking no" "${UPGRADE_USER}@${CP_IP}" -- kubectl apply -f https://docs.projectcalico.org/manifests/calico.yaml
+
+provision_worker_node
+wait_for_worker_provisioning_start
+
+ORIGINAL_NODE=$(kubectl get bmh -n metal3 | grep worker | grep -v ready | cut -f1 -d' ')
+NODE_IP=$(sudo virsh net-dhcp-leases baremetal | grep "${ORIGINAL_NODE}" | awk '{{print $5}}' | cut -f1 -d'/')
+
+wait_for_worker_provisioning_complete 4 "${ORIGINAL_NODE}" "${NODE_IP}" "Worker node"
+
+deploy_workload_on_workers
+
+manage_node_taints "${CP_IP}"
+
+wait_for_node_to_scale_to 0 "${CP_IP}" "worker"
+
+# k8s version upgrade
+FROM_VERSION=$(kubectl get kcp -n metal3 -oyaml | grep "version: v1" | cut -f2 -d':' | awk '{$1=$1;print}')
+
+if [[ "${FROM_VERSION}" < "${UPGRADED_K8S_VERSION_2}" ]]; then
+  TO_VERSION="${UPGRADED_K8S_VERSION_2}"
+elif [[ "${FROM_VERSION}" > "${KUBERNETES_VERSION}" ]]; then
+  TO_VERSION="${KUBERNETES_VERSION}"
+else
+  exit 0
+fi
+
+# Node image version upgrade
+M3_MACHINE_TEMPLATE_NAME=$(kubectl get Metal3MachineTemplate -n metal3 -oyaml | grep "name: " | grep controlplane | cut -f2 -d':' | awk '{$1=$1;print}')
+
+Metal3MachineTemplate_OUTPUT_FILE="/tmp/new_image.yaml"
+CLUSTER_UID=$(kubectl get clusters -n metal3 "${CLUSTER_NAME}" -o json | jq '.metadata.uid' | cut -f2 -d\")
+IMG_CHKSUM=$(curl -s https://cloud-images.ubuntu.com/bionic/current/MD5SUMS | grep bionic-server-cloudimg-amd64.img | cut -f1 -d' ')
```

Comment:
Please refer to my previous comment with regards to fetching the checksum. you already have it on disk